Src/Screens/OfferScreen/Screens/PresentStartDate.swift

```diff
@@ -0,0 +1,177 @@
+
+        containerView.addArrangedSubview(pickerStackView)
+        
+        let actionStackView = UIStackView()
+        actionStackView.spacing = 24
+        actionStackView.axis = .vertical
+        actionStackView.alignment = .center
+        actionStackView.layoutMargins = UIEdgeInsets(top: 32, left: 0, bottom: 10, right: 0)
+        actionStackView.isLayoutMarginsRelativeArrangement = true
+        actionStackView.isUserInteractionEnabled = true
+        
+        containerView.addArrangedSubview(actionStackView)
+
+        let titleLabel = MultilineLabel(
+            value: String("Byt startdatum"),
+            style: .startDateTitle
+        )
+      
+        bag += textStackView.addArranged(titleLabel)
+
+        let descriptionLabel = MultilineLabel(
+            value: String("Vilket datum vill du att din försäkring aktiveras?"),
+            style: .startDateDescription
+        )
+        
+        bag += textStackView.addArranged(descriptionLabel)
+        
+        let picker = UIDatePicker()
+        
+        picker.calendar = Calendar.current
+        picker.datePickerMode = .date
+        picker.minimumDate = Date()
+        picker.maximumDate = Calendar.current.date(byAdding: .year,
+                                                   value: 1,
+                                                   to: Date())
+        
+        let dateFormatter = DateFormatter()
+        dateFormatter.dateFormat = "yyyy-MM-dd"
+        dateFormatter.timeZone = NSTimeZone.local
+        let today = dateFormatter.string(from: Date())
+        
+        var valueSelected: String = today
+        
+        bag += picker.atOnce().onValue({ (_) in
+            let pickedValue = dateFormatter.string(from: picker.value)
+            valueSelected = pickedValue
+            print("PICKED VALUE: \(valueSelected)")
+        })
+
+        pickerStackView.addArrangedSubview(picker)
+        
+        let chooseDateButton = Button(title: "Välj datum",
+                                      type: .standard(backgroundColor: .primaryTintColor,
+                                                      textColor: .white))
+        
+        let loadableChooseDateButton = LoadableButton(button: chooseDateButton,
+                                                  initialLoadingState: false)
+        
+        let activateNowButton = Button(title: "",
+                                       type: .transparent(textColor: .primaryTintColor))
+        
+        let loadableActivateButton = LoadableButton(button: activateNowButton,
+                                                    initialLoadingState: false)
+        
+        bag += actionStackView.addArranged(loadableChooseDateButton.wrappedIn(UIStackView()))
+        bag += actionStackView.addArranged(loadableActivateButton.wrappedIn(UIStackView()))
+        
+        
+        let innerbag = bag.innerBag()
+        
+        bag += self.client.fetch(query: HasPreviousInsuranceQuery()).map{
```

Comment:
You shouldn't do operations like these in a `.map` you should use `onValue` instead.